river/tree/mondrian/mondrian_tree_classifier.py

```diff
@@ -311,23 +312,26 @@ def _go_downwards(self):
             # Path from the parent to the current node
             branch_no = None
             while True:
+                # Computing the extensions to get the intensities
+                extensions_sum, extensions = current_node.range_extension(self._x)
+
                 # If it's not the first iteration (otherwise the current node
                 # is root with no range), we consider the possibility of a split
-                split_time = self._compute_split_time(current_node)
+                split_time = self._compute_split_time(current_node, extensions_sum)
 
                 if split_time > 0:
                     # We split the current node: because the current node is a
                     # leaf, or because we add a new node along the path
 
-                    # Computing the extensions to get the intensities
-                    extensions_sum, extensions = current_node.range_extension(self._x)
                     # We normalize the range extensions to get probabilities
-                    intensities = utils.norm.normalize_values_in_dict(extensions, inplace=True)
+                    intensities = utils.norm.normalize_values_in_dict(extensions, inplace=False)
 
                     # Sample the feature at random with a probability
                     # proportional to the range extensions
+
+                    candidates = sorted(list(self._x.keys()))
```

Comment:
That's why ordering was influencing the results. Now I'm sorting the keys to ensure consistency

Reply:
aaahw makes sense xD